src/ansible_compat/runtime.py

```diff
@@ -772,18 +776,16 @@ def require_collection(
                             _logger.fatal(msg)
                             raise InvalidPrerequisiteError(msg)
                     return found_version, collpath.resolve()
-                break
-        else:
```

Comment:
It doesn't look like this `break` can ever be reached, which means the else is also superfluous as anything that exits the loop does so by exhausting `paths`